### ISSUE TITLE
JOSS: documentation https://opencmp.io/

### ISSUE THREAD

**Issue by bonh** [uw-comphys/opencmp#18]
While reading through your documentation I will add notes for further discussion here.

**Comment by bonh** [comment]
https://opencmp.io/getting_started/installation_guide.html, add references or at least links to the dependencies.

**Comment by bonh** [comment]
> Optionally, run the unit tests `python -m pytests` from the top-level directory.
Perhaps you mean `python -m pytest`?

**Comment by bonh** [comment]
executing `python -m pytest` eats up all my resources. Perhaps you should add a flag or something to limit cores or threads (not sure how)?

**Comment by bonh** [comment]
The test run for a really long time; perhaps you should add a warning to the text or something.

**Comment by Alex-Vasile** [comment]
The tests now use only a single thread to run. To run the slow tests, the flag `--runslow` must be used. The README.md inside pytests/ has further info about this.

[Documentation](https://opencmp.io/getting_started/installation_guide.html) was also updated.

**Comment by bonh** [comment]
Thank you, that is helpful! 

What about that one?

> https://opencmp.io/getting_started/installation_guide.html, add references or at least links to the dependencies.

**Comment by bonh** [comment]
`SKIPPED [1] pytests/full_system/ins/test_ins.py:79: need --runslow option to run`, perhaps explain `runslow` in the installation guide https://opencmp.io/getting_started/installation_guide.html?

And I can start the tests only with `python3 -m pytest -runslow`, so only one dash before runslow?

**Comment by bonh** [comment]
Mhh, no, most likely I am doing something wrong. What is the command to run all the tests at once (including the slow tests)?

**Comment by bonh** [comment]
From the `README` inside `pytests`: `pytest --runslow` but that results in `ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --runslow
  inifile: None
  rootdir: /root/opencmp`

I am using a docker with ubuntu and installed everything using apt and pip3.

**Comment by bonh** [comment]
Perhaps it might be best (and more easier for me) if you could provide a docker image with everything one needs to use opencmp preinstalled?

**Comment by nasserma** [comment]
> Thank you, that is helpful!
> 
> What about that one?
> 
> > https://opencmp.io/getting_started/installation_guide.html, add references or at least links to the dependencies.

We modified this page as well to provide a link to the NGSolve installation page and information required to install the other dependencies using pip. Are there additional changes that you would like us to add?

**Comment by nasserma** [comment]
> `SKIPPED [1] pytests/full_system/ins/test_ins.py:79: need --runslow option to run`, perhaps explain `runslow` in the installation guide https://opencmp.io/getting_started/installation_guide.html?
> 
> And I can start the tests only with `python3 -m pytest -runslow`, so only one dash before runslow?

We will take a look at this, the --runslow option does not seem to be consistently implemented (should work using pytest script, but does not work using python -m pytest).

**Comment by bonh** [comment]
> > Thank you, that is helpful!
> > What about that one?
> > > https://opencmp.io/getting_started/installation_guide.html, add references or at least links to the dependencies.
> 
> We modified this page as well to provide a link to the NGSolve installation page and information required to install the other dependencies using pip. Are there additional changes that you would like us to add?

I am just suggesting that you add references, for example from that page https://ngsolve.org/documentation/publications. And links to https://github.com/pyparsing/pyparsing, https://github.com/seung-lab/euclidean-distance-transform-3d/. And any other links to important software or references. To give credit, don't you agree?

**Comment by Alex-Vasile** [comment]
> From the `README` inside `pytests`: `pytest --runslow` but that results in `ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...] pytest: error: unrecognized arguments: --runslow inifile: None rootdir: /root/opencmp`
> 
> I am using a docker with ubuntu and installed everything using apt and pip3.

Yes, sorry about that, it is a typo. `-runslow` works as intented.

**Comment by Alex-Vasile** [comment]
> `SKIPPED [1] pytests/full_system/ins/test_ins.py:79: need --runslow option to run`, perhaps explain `runslow` in the installation guide https://opencmp.io/getting_started/installation_guide.html?
> 
> And I can start the tests only with `python3 -m pytest -runslow`, so only one dash before runslow?

We have a description of `-runslow` in the README.md inside pytests/, which the installation guide points users towards for more information about running the tests.

**Comment by nasserma** [comment]
We have made changes to our documentation to account for issues with the --runslow option using the pytest module and, based on you indicating that you were successfully able to run all tests here,

https://github.com/uw-comphys/opencmp/issues/19#issuecomment-1030589241

I am closing this issue.

**Comment by bonh** [comment]
What about this? Please engage in a discussion with me - I am trying to help here!

>I am just suggesting that you add references, for example from that page >https://ngsolve.org/documentation/publications. And links to https://github.com/pyparsing/pyparsing, >https://github.com/seung-lab/euclidean-distance-transform-3d/. And any other links to important software or >references. To give credit, don't you agree?

**Comment by bonh** [comment]
Running `python3 visualization` from https://opencmp.io/tutorials/tutorial_9.html requires `matplotlib`, right?

**Comment by bonh** [comment]
On https://opencmp.io/tutorials/tutorial_9.html it seems the equations are missing?

**Comment by nasserma** [comment]
Yes, this seems to be the case, we'll correct this shortly.

**Comment by bonh** [comment]
> What about this? Please engage in a discussion with me - I am trying to help here!

> I am just suggesting that you add references, for example from that page >https://ngsolve.org/documentation/publications. And links to https://github.com/pyparsing/pyparsing, >https://github.com/seung-lab/euclidean-distance-transform-3d/. And any other links to important software or >references. To give credit, don't you agree?

What about adding references or/and links to the packages you are using?

**Comment by nasserma** [comment]
We added the lines for installing these packages using PIP in the installation guide,

https://opencmp.io/getting_started/installation_guide.html

so with the unique identifier of each package the user could get all info from "pip show IDENTIFIER". If you want us to also include links directory to their websites, I'll do that shortly, however I now realize that it would be better to determine if any of these packages have published papers associated with them and to cite them in the JOSS submission.

**Comment by bonh** [comment]
Well, I was merely suggesting giving credit, yes. Sure, I can always try to find everything myself but I appreciate if developers explicitly acknowledge other peoples work. So in your documentation, I think it would be beneficial to add a link to the github repos, for example. Citing published papers in you own paper is even better.
